include/lsst/daf/base/PropertySet.h

```diff
@@ -85,83 +76,371 @@ class PropertySet : public Persistable, public Citizen {
+     * @throws TypeError Type does not match existing values.
+     * @throws InvalidParameterError Hierarchical name uses non-PropertySet.
+     */
     template <typename T> void add(std::string const& name, T const& value);
+
+    /**
+     * Append a vector of values to the vector of values for a property name
+     * (possibly hierarchical).  Sets the values if the property does not exist.
+     *
+     * @param[in] name Property name to append to, possibly hierarchical.
+     * @param[in] value Vector of values to append.
+     * @throws TypeError Type does not match existing values.
+     * @throws InvalidParameterError Hierarchical name uses non-PropertySet.
+     * @note
+     * May only partially add the vector if an exception occurs.
+     */
     template <typename T> void add(std::string const& name,
                                    std::vector<T> const& value);
+
+    /**
+     * Append a <tt>char const*</tt> value to the vector of values for a
+     * property name (possibly hierarchical).  Sets the value if the property
+     * does not exist.
+     *
+     * @param[in] name Property name to append to, possibly hierarchical.
+     * @param[in] value Character string value to append.
+     * @throws TypeError Type does not match existing values.
+     * @throws InvalidParameterError Hierarchical name uses non-PropertySet.
+     */
     void add(std::string const& name, char const* value);
 
-    virtual void copy(std::string const& dest, ConstPtr source,
-                      std::string const& name);
+    /**
+     * Replace a single value vector in the destination with one from the
+     * \a source.
+     *
+     * @param[in] dest Destination property name.
+     * @param[in] source PropertySet::Ptr for the source PropertySet.
+     * @param[in] name Property name to extract.
+     * @param[in] asScalar If true copy the item as a scalar by ignoring all but the last value
+     *                     (which is the value returned by get<T>(name))
+     * @throws TypeError Type does not match existing values.
+     * @throws InvalidParameterError Name does not exist in source.
+     * @throws InvalidParameterError Hierarchical name uses non-PropertySet.
+     */
+    virtual void copy(std::string const& dest, ConstPtr source, std::string const& name,
+                      bool asScalar = false);
+
+    /**
+     * Append all value vectors from the \a source to their corresponding
+     * properties.  Sets values if a property does not exist.
+     *
+     * @param[in] source PropertySet::Ptr for the source PropertySet.
+     * @throws TypeError Type does not match existing values.
+     * @throws InvalidParameterError Hierarchical name uses non-PropertySet.
+     * @note
+     * May only partially combine the PropertySets if an exception occurs.
+     */
     virtual void combine(ConstPtr source);
-        // All vectors from the source are add()ed to the destination with the
-        // same names.  Types must match.
```

Comment:
Should this ("types must match") move up to the doxygen comment instead of being removed?  Or is the `@throws TypeError` sufficient?

Reply:
I'll add a note to the doc.